> Without crawling the headers to check for sure, I dont think this is truly necessary. We've linked and run the plugin without the gazebo libraries and if you run `ldd` on the library it doesn't show up, so it hasn't even been accidentally included as a transitive dependency.

When I don't link in neither gazebo nor protobuf (i.e. without this patch), I get:

```
Linking CXX shared library /Users/demmeln/work/test-ws/devel/lib/libgazebo_ros_kobuki.dylib
Undefined symbols for architecture x86_64:
  "sdf::Console::ColorMsg(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int, int)", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Param::Get<double>(double&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Param::Get<bool>(bool&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Param::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >&) in gazebo_ros_kobuki.cpp.o
  "sdf::Console::Instance()", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Param::Get<double>(double&) in gazebo_ros_kobuki.cpp.o
      sdf::Console::ConsoleStream& sdf::Console::ConsoleStream::operator<<<char [30]>(char const (&) [30]) in gazebo_ros_kobuki.cpp.o
      sdf::Console::ConsoleStream& sdf::Console::ConsoleStream::operator<<<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      sdf::Console::ConsoleStream& sdf::Console::ConsoleStream::operator<<<char [3]>(char const (&) [3]) in gazebo_ros_kobuki.cpp.o
      ...
  "sdf::Element::GetElement(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "sdf::Element::GetAttribute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
  "sdf::Element::HasElementDescription(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Quaternion::GetYaw()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Quaternion::~Quaternion()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Pose::Pose(gazebo::math::Pose const&)", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Pose::~Pose()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Angle::~Angle()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Vector3::Vector3()", referenced from:
      gazebo::GazeboRosKobuki::GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Vector3::~Vector3()", referenced from:
      gazebo::GazeboRosKobuki::GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::~GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Vector3::operator=(gazebo::math::Vector3 const&)", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::msgs::Contacts::CopyFrom(gazebo::msgs::Contacts const&)", referenced from:
      gazebo::msgs::Contacts::operator=(gazebo::msgs::Contacts const&) in gazebo_ros_kobuki.cpp.o
  "gazebo::msgs::Contacts::Contacts()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::msgs::Contacts::~Contacts()", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::event::Connection::Connection(gazebo::event::Event*, int)", referenced from:
      gazebo::event::EventT<void (gazebo::common::UpdateInfo const&)>::Connect(boost::function<void (gazebo::common::UpdateInfo const&)> const&) in gazebo_ros_kobuki.cpp.o
  "gazebo::event::Connection::~Connection()", referenced from:
      void boost::checked_delete<gazebo::event::Connection>(gazebo::event::Connection*) in gazebo_ros_kobuki.cpp.o
  "gazebo::event::Events::worldUpdateBegin", referenced from:
      boost::shared_ptr<gazebo::event::Connection> gazebo::event::Events::ConnectWorldUpdateBegin<boost::_bi::bind_t<void, boost::_mfi::mf0<void, gazebo::GazeboRosKobuki>, boost::_bi::list1<boost::_bi::value<gazebo::GazeboRosKobuki*> > > >(boost::_bi::bind_t<void, boost::_mfi::mf0<void, gazebo::GazeboRosKobuki>, boost::_bi::list1<boost::_bi::value<gazebo::GazeboRosKobuki*> > >) in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Time::Time()", referenced from:
      gazebo::GazeboRosKobuki::GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Time::~Time()", referenced from:
      gazebo::GazeboRosKobuki::GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::~GazeboRosKobuki() in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::cmdVelCB(boost::shared_ptr<geometry_msgs::Twist_<std::__1::allocator<void> > const> const&) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Time::operator=(gazebo::common::Time const&)", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::cmdVelCB(boost::shared_ptr<geometry_msgs::Twist_<std::__1::allocator<void> > const> const&) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Console::ColorMsg(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Console::Console()", referenced from:
      SingletonT<gazebo::common::Console>::GetInstance() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Console::~Console()", referenced from:
      SingletonT<gazebo::common::Console>::GetInstance() in gazebo_ros_kobuki.cpp.o
  "gazebo::physics::Model::GetJoint(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::SensorManager::SensorManager()", referenced from:
      SingletonT<gazebo::sensors::SensorManager>::GetInstance() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::SensorManager::~SensorManager()", referenced from:
      SingletonT<gazebo::sensors::SensorManager>::GetInstance() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::RaySensor::GetRange(int)", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "google::protobuf::internal::LogMessage::LogMessage(google::protobuf::LogLevel, char const*, int)", referenced from:
      google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
      google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
  "google::protobuf::internal::LogMessage::~LogMessage()", referenced from:
      google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
      google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
  "google::protobuf::internal::LogMessage::operator<<(char const*)", referenced from:
      google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
      google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
  "google::protobuf::internal::LogFinisher::operator=(google::protobuf::internal::LogMessage&)", referenced from:
      google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Vector3d>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
      google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler::Type const& google::protobuf::internal::RepeatedPtrFieldBase::Get<google::protobuf::RepeatedPtrField<gazebo::msgs::Contact>::TypeHandler>(int) const in gazebo_ros_kobuki.cpp.o
  "sdf::Element::HasElement(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
  "sdf::Element::GetElementImpl(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
  "sdf::Element::GetElementDescription(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > sdf::Element::Get<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      bool sdf::Element::Get<bool>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
      double sdf::Element::Get<double>(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) in gazebo_ros_kobuki.cpp.o
  "sdf::Element::GetParent() const", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "gazebo::math::Angle::Radian() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Time::Double() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::common::Time::operator-(gazebo::common::Time const&) const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::physics::Base::GetWorld() const", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "gazebo::physics::Joint::GetAngle(int) const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::physics::World::GetSimTime() const", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::cmdVelCB(boost::shared_ptr<geometry_msgs::Twist_<std::__1::allocator<void> > const> const&) in gazebo_ros_kobuki.cpp.o
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::ContactSensor::GetContacts() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::SensorManager::GetSensor(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const", referenced from:
      gazebo::GazeboRosKobuki::Load(boost::shared_ptr<gazebo::physics::Model>, boost::shared_ptr<sdf::Element>) in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::ImuSensor::GetOrientation() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::ImuSensor::GetAngularVelocity() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "gazebo::sensors::ImuSensor::GetLinearAcceleration() const", referenced from:
      gazebo::GazeboRosKobuki::OnUpdate() in gazebo_ros_kobuki.cpp.o
  "typeinfo for gazebo::sensors::ContactSensor", referenced from:
      boost::shared_ptr<gazebo::sensors::ContactSensor> boost::dynamic_pointer_cast<gazebo::sensors::ContactSensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
  "typeinfo for gazebo::sensors::Sensor", referenced from:
      boost::shared_ptr<gazebo::sensors::RaySensor> boost::dynamic_pointer_cast<gazebo::sensors::RaySensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
      boost::shared_ptr<gazebo::sensors::ContactSensor> boost::dynamic_pointer_cast<gazebo::sensors::ContactSensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
      boost::shared_ptr<gazebo::sensors::ImuSensor> boost::dynamic_pointer_cast<gazebo::sensors::ImuSensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
  "typeinfo for gazebo::sensors::ImuSensor", referenced from:
      boost::shared_ptr<gazebo::sensors::ImuSensor> boost::dynamic_pointer_cast<gazebo::sensors::ImuSensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
  "typeinfo for gazebo::sensors::RaySensor", referenced from:
      boost::shared_ptr<gazebo::sensors::RaySensor> boost::dynamic_pointer_cast<gazebo::sensors::RaySensor, gazebo::sensors::Sensor>(boost::shared_ptr<gazebo::sensors::Sensor> const&) in gazebo_ros_kobuki.cpp.o
```

--- a/kobuki_gazebo_plugins/CMakeLists.txt
+++ b/kobuki_gazebo_plugins/CMakeLists.txt
@@ -1,11 +1,7 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(kobuki_gazebo_plugins)
 
-# Find gazebo
-include (FindPkgConfig)
-if (PKG_CONFIG_FOUND)
-  pkg_check_modules(GAZEBO gazebo)
-endif()
+find_package(gazebo REQUIRED)
 
 find_package(catkin REQUIRED COMPONENTS gazebo_ros
                                         geometry_msgs
@@ -38,7 +34,9 @@ add_dependencies(gazebo_ros_kobuki geometry_msgs_gencpp
                                    nav_msgs_gencpp
                                    sensor_msgs_gencpp
                                    std_msgs_gencpp)
-target_link_libraries(gazebo_ros_kobuki ${catkin_LIBRARIES})
+target_link_libraries(gazebo_ros_kobuki
+                      ${catkin_LIBRARIES}
+                      ${GAZEBO_LIBRARIES})
 
 install(TARGETS gazebo_ros_kobuki
         DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})

